packages/flutter_goldens/lib/src/flutter_goldens_io.dart

```diff
@@ -20,28 +22,110 @@ export 'package:flutter_goldens_client/skia_client.dart';
 /// [goldenFileComparator] to an instance of [FlutterGoldenFileComparator] that
 /// works for the current test. _Which_ FlutterGoldenFileComparator is
 /// instantiated is based on the current testing environment.
 ///
 /// When set, the `namePrefix` is prepended to the names of all gold images.
 Future<void> testExecutable(FutureOr<void> Function() testMain, {String? namePrefix}) async {
+  _namePrefix = namePrefix;
+  final GoldenFileComparator nonFlakyComparator = await _createComparator();
+  _flakyGoldenFileComparator = await _createComparator(isFlaky: true);
+
+  // Assign this last because FlutterLocalFileComparator.fromDefaultComparator
+  // is called twice, and it needs the original value of [goldenFileComparator].
+  goldenFileComparator = nonFlakyComparator;
+  await testMain();
+}
+
+/// Processes golden check commands sent from the browser process.
+///
+/// When running browser tests, goldens are not generated within the app itself
+/// due to browser restrictions. Instead, when a test calls [matchesFlutterGolden]
+/// the browser sends a [command] to a host process. This function handles the
+/// command.
+///
+/// This custom command handler is used for Flutter's own goldens. It
+/// understands the "isFlaky" property, a boolean encoded in the command as a
+/// custom command property. If true, uses a golden comparator that submits the
+/// golden, but does not fail the test, allowing manual inspection in the Skia
+/// Gold UI and verification of fixes.
+///
+/// See also:
+///   * [FlakyWebGoldenComparator], which implements custom browser-side logic.
+Future<void> processBrowserCommand(dynamic command) async {
+  if (command is Map<String, dynamic>) {
+    final io.File imageFile = io.File(command['imageFile'] as String);
+    final Uri goldenKey = Uri.parse(command['key'] as String);
+    final bool update = command['update'] as bool;
+    final Map<String, dynamic> customProperties = (command['customProperties'] as Map<String, dynamic>?) ?? const <String, dynamic>{};
+    final bool isFlaky = (customProperties['isFlaky'] as bool?) ?? false;
+    final Uint8List bytes = await io.File(imageFile.path).readAsBytes();
+    if (update) {
+      await goldenFileComparator.update(goldenKey, bytes);
+      io.stdout.writeln(jsonEncode(<String, dynamic>{'success': true}));
+    } else {
+      try {
+        final bool success = isFlaky
+          ? await _flakyGoldenFileComparator.compare(bytes, goldenKey)
+          : await goldenFileComparator.compare(bytes, goldenKey);
+        io.stdout.writeln(jsonEncode(<String, dynamic>{'success': success}));
+      } on Exception catch (exception) {
+        io.stdout.writeln(jsonEncode(<String, dynamic>{'success': false, 'message': '$exception'}));
+      }
+    }
+  } else {
+    io.stdout.writeln('object type is not right');
+  }
+}
+
+Future<GoldenFileComparator> _createComparator({ bool isFlaky = false }) async {
```

Comment:
Why does the comparator need to know if something is flaky? Couldn't it just be a flag on `compare`? IIRC, we create a comparator for each test file, not each individual test.

Reply:
That was my proposal [in the previous PR](https://github.com/flutter/flutter/pull/111595/files#diff-837c442b591a864e039d7340eaddbe923b8c0b0305f8ee39fbe9290d5b2f4d9bR75), but that change turned out to be breaking and it made the `isFlaky` flag public, both aspects considered undesirable in review comments. However, I'm not against that approach. It will just require a small migration.

Reply:
Sorry I meant the `compare` of the super class of Flutter's specific comparator (`FlutterGoldenFileComparator`).. not the public `GoldenFileComparator.compare`. Can we not add isFlaky to that comparison path?

Reply:
Or a new FlutterGoldenFileComparator.compareFlaky method?

Reply:
Oh, yeah, sub-classes can add extra parameters. I'll give it a shot.

Reply:
This turned out not so simple: https://github.com/flutter/flutter/pull/113396#discussion_r1001199326